src/main/java/com/zendesk/maxwell/MaxwellAbstractRowsEvent.java

```diff
@@ -196,6 +197,22 @@ public String toSQL() {
 	class RowMap extends HashMap<String, Object> {
 		private final HashMap<String, Object> data;
 
+		@Override
+		public Set<String> keySet() {
```

Comment:
Is this in the hot loop?

Seems like `keySet` only has 4 possible values that are basically immutable

Constructing objects in Java creates garbage, increased GC pressure, etc.

Could we create four LinkedHashSets:

```
keySets[0] = database, table, type, ts
keySets[1] = database, table, type, ts, xid
keySets[2] = database, table, type, ts, commit
keySets[3] = database, table, type, ts, xid, commit
```
(expand to needed Java boilerplate)

Then

```
int xidOffset = this.containsKey("xid") ? 1 : 0;
inf commitOffset = (this.containsKey("commit") && (boolean) this.get("commit") == true)) ? 2 : 0;
return keySets[xidOffset + commitOffset];
```